pytensor/link/pytorch/dispatch/elemwise.py

```diff
@@ -0,0 +1,71 @@
+import torch
+
+from pytensor.link.pytorch.dispatch.basic import pytorch_funcify
+from pytensor.tensor.elemwise import CAReduce, DimShuffle, Elemwise
+from pytensor.tensor.special import LogSoftmax, Softmax, SoftmaxGrad
+
+
+@pytorch_funcify.register(Elemwise)
+def pytorch_funcify_Elemwise(op, node, **kwargs):
+    scalar_op = op.scalar_op
+    base_fn = pytorch_funcify(scalar_op, node=node, **kwargs)
+
+    def elemwise_fn(*inputs):
+        Elemwise._check_runtime_broadcast(node, tuple(map(torch.tensor, inputs)))
```

Comment:
Shouldn't the inputs be `torch.tensor`s already?

Reply:
Yeah this looks copied over from JAX impl where we need to get around the fact we tipyfied stuff into python integers and floats as you mentioned above. Let's try without the hack and the need to re-wrap

Reply:
I think an extra condition like:
```
elif isinstance(data, torch.Tensor):
            return data
```
can be added here (in class `TensorType`):
https://github.com/pymc-devs/pytensor/blob/d80c0bf7bfa5d6b9812ef6381cfecb6b9bf948a6/pytensor/tensor/type.py#L141-L162
This will make sure `torch.tensor`s are not converted to numpy